pkg/executables/kind.go

```diff
@@ -134,6 +135,22 @@ func (k *Kind) WithExtraDockerMounts() bootstrapper.BootstrapClusterClientOption
 	}
 }
 
+func (k *Kind) WithExtraPortMappings(ports []int) bootstrapper.BootstrapClusterClientOption {
+	return func() error {
+		if k.execConfig == nil {
+			return errors.New("kind exec config is not ready")
```

Comment:
Should this panic? Its not like a user can do anything about this and it feels very much like a developer has made a significant error that's rendered the program invalid.

Reply:
I just followed what all the other kind options are doing. If we decide to panic instead, that would be a broader change and I would prefer to do that in a separate PR